netbox/dcim/models/racks.py

```diff
@@ -29,9 +29,183 @@
     'Rack',
     'RackReservation',
     'RackRole',
+    'RackType',
 )
 
 
+#
+# Rack Types
+#
+
+class RackType(WeightMixin, PrimaryModel):
+    """
+    Devices are housed within Racks. Each rack has a defined height measured in rack units, and a front and rear face.
+    Each Rack is assigned to a Site and (optionally) a Location.
+    """
+    RACK_FIELDS = [
+        'type', 'width', 'u_height', 'starting_unit', 'desc_units', 'outer_width',
+        'outer_depth', 'outer_unit', 'weight', 'weight_unit', 'max_weight',
+        'mounting_depth'
+    ]
+
+    manufacturer = models.ForeignKey(
+        to='dcim.Manufacturer',
+        on_delete=models.PROTECT,
+        related_name='rack_types'
+    )
+    name = models.CharField(
+        verbose_name=_('name'),
+        max_length=100
+    )
+    _name = NaturalOrderingField(
+        target_field='name',
+        max_length=100,
+        blank=True
+    )
+    slug = models.SlugField(
+        verbose_name=_('slug'),
+        max_length=100,
+        unique=True
+    )
+    type = models.CharField(
```

Comment:
IMO we should rename `type` to avoid confusion with RackType. Maybe `form_factor`, or something similar?

Reply:
I've renamed it to `form_factor`.